crossdock/client/client.go

```diff
@@ -36,19 +38,24 @@ func behaviorRequestHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// All behaviors will eventually contribute to this behavior's output.
-	bt := BehaviorTester{Params: httpParams{Request: r}}
-	v := bt.Param("behavior")
-	switch v {
-	case "raw", "json", "thrift":
-		runEchoBehavior(&bt, v)
-	default:
-		bt.NewBehavior(BasicEntryBuilder).Skipf("unknown behavior %v", v)
+	var s behavior.EntrySink
+	behavior.Run(func() { dispatch(&s, httpParams{r}) })
```

Comment:
:+1: 

To make that happen, I had to change `Run` to have information about the sink so the interface is now:

    Run(func(Sink)) []interface{} { ... }

So Run construct the sink and returns the entries logged by the function, and it handles panics.